docs/design/fabric-v2+/fpc-validation.puml

```diff
@@ -43,97 +37,127 @@ ref over User1, Orderer
 end ref
 
 group endorsement
-    activate User1
+    activate CLI
 
-    User1 -> ECC1 ++: send proposal
-    ref over ECC1
-        invoke chaincode enclave
+    User1 -> CLI : send proposal
+    ref over CLI, ECC1, ECC2, ECC3
+        invoke chaincode enclave and collect enough enclave endorsements { proposal-response }
+        and corresponding results { ccrsp_msg } to satisfy enclave endorsement policy
         see fpc-cc-invocation.puml
     end ref
-    return proposal response
 
+    alt FPC "light"
+        group enclave endorsement validation transaction
+            CLI -> ECC1 ++: invoke "validateEnclaveEndorsement" ({ ccrsp_msg })
+                group enclave endorsement validation
+                    ECC1 -> ECC1 : <ECC_result, read/writeset, proposal, Sig_Enclave, Enclave_Id> <- extract from  ccrsp_msg
+  
+                    create participant ERCC order 30
+                    ECC1 -> ERCC++ : queryEnclaveCredentials(CC_Id, Enclave_Id)
+                        note right
+                            If FPC enclave is registered, ERCC will
+                            return the corresponding Credentials,
+                            otherwise Credentials is empty and validation will abort.
+                        end note
+                    return Credentials
+                    ECC1 -> ECC1 : extract Enclave_VK from Credentials.attesated_data
+                    ECC1 -> ECC1 : verify Sig_Enclave over <proposal, read/writeset, ECC_result> with Enclave_VK
+                    ECC1 -> ECC1 : extract CC_Params from Credentials.AttestedData
+                    ECC1 -> ECC1 : check CC_Params matches chaincode definition
```

Comment:
```suggestion
                    ECC1 -> ECC1 : check CC_Params matches chaincode definition stored in Credentials
```

Reply:
`CC_Params` are already extracted from `Credentials`, so that test would always be true. What we have to test that CC_Param matches ECCs view of the chaincode definition. I've handled that by adding a "own" after "matches".

PS: made me realize that we somewhat mix up peer and fpc-stub here, maybe also leading to your confusion below regarding fpc validation. Split now the two functionalities.